### DISCUSSION TITLE
Enable boundary conditions ONLY on one side of the geometry/domain/grid

### DISCUSSION THREAD

**Discussion by yibaoj** [zwicker-group/py-pde#494] [Q&A]
I am solving a simple 1-D time-dependent PDE as follow ($a$ and $b$ are pre-defined coefficients):
$$\frac{\partial u}{\partial t} = a \frac{\partial u}{\partial x} + b T,\ t \in [0, T],\ x \in [0, L].$$

The initial and boundary conditions are given by:
$$u(t=0) = f(x), u(x=0) = g(t).$$

In simplest form, $f(x)$ and $g(t)$ can be constants.

Following is my code snippet. However, I couldn't find a easy way to implement the boundary conditions ONLY on **one side** of axis $x$, namely $x=0$. Consequently, results went diverged.

Do you have any suggestions to fix the problem?

```
T, L = 300, 150
a, b = 0.5, 0.002
grid = pde.CartesianGrid([[0, L]], [100], periodic=False)
state = pde.ScalarField(grid, data=80)
bc = [{"value": 80}, ]
eq = pde.PDE({"u": f"- {a} * d_dx(u) - {b} * u"}, bc=bc) 
print(eq.expressions)
sol = eq.solve(state, t_range=T, dt=10, tracker=pde.PlotTracker())
```
<img width="335" alt="image" src="https://github.com/zwicker-group/py-pde/assets/24381205/aeffb607-4459-4ce0-80e6-4b62b4005367">
<img width="340" alt="image" src="https://github.com/zwicker-group/py-pde/assets/24381205/282fbd87-cb8f-40d7-b0d7-bdb7e8d66ad5">

**Comment by david-zwicker** [comment]
The derivative operator `d_dx` by default uses a central derivative, so both boundary conditions need to be set. For advective problems like yours, one-sided derivatives can be more appropriate. You can construct one-sided derivatives using `d_dx_forward` and `d_dx_backward` if you use the `master` branch of the package. Discussions #312 and #426 have some additional information on advection equations.
